### PR TITLE
5 Parental Bond tests

### LINKED ISSUE #3430: Write missing Parental Bond tests
### Description

#3417 required merging, but I only made a single test to make sure that the fix worked. The rest of the tests need to be made.

### Version

1.6.1 (Default)

### Upcoming Version

_No response_

### Discord contact info

AsparagusEduardo

### PR DESCRIPTION
**Pull request by Bassoonian**
Adds five of the many Parental Bond tests that still need to be written

## Issue(s) that this PR fixes
Partially fixes #3430 

## **Discord contact info**
bassoonian

### CODE REVIEW

File: test/battle/ability/parental_bond.c [hunk 1]
```diff
@@ -24,23 +24,166 @@ SINGLE_BATTLE_TEST("Parental Bond converts Tackle into a two-strike move")
     }
 }
 
+SINGLE_BATTLE_TEST("Parental Bond does not convert Triple Kick into a two-strike move")
+{
+    GIVEN {
+        ASSUME(gBattleMoves[MOVE_TRIPLE_KICK].split != SPLIT_STATUS);
+        ASSUME(gBattleMoves[MOVE_TRIPLE_KICK].strikeCount == 3);
+        ASSUME(gBattleMoves[MOVE_TRIPLE_KICK].effect == EFFECT_TRIPLE_KICK);
```
Comment (AlexOn1ine) [inline test/battle/ability/parental_bond.c:32]: Is the assume for the effect needed here it shouldn't effect the test in any way?

File: test/battle/ability/parental_bond.c [hunk 2]
```diff
@@ -24,23 +24,166 @@ SINGLE_BATTLE_TEST("Parental Bond converts Tackle into a two-strike move")
...
+        EXPECT_EQ(player->species, SPECIES_KANGASKHAN_MEGA);
+    }
+}
+
+SINGLE_BATTLE_TEST("Parental Bond converts multi-target moves into a two-strike move in Single Battles")
+{
+    GIVEN {
+        ASSUME(gBattleMoves[MOVE_EARTHQUAKE].split != SPLIT_STATUS);
+        ASSUME(gBattleMoves[MOVE_EARTHQUAKE].strikeCount < 2);
+        ASSUME(gBattleMoves[MOVE_EARTHQUAKE].effect == EFFECT_EARTHQUAKE);
```
Comment (AlexOn1ine) [inline test/battle/ability/parental_bond.c:56]: Again not sure if assuming Earthquake effect matters here but I think the assume for `MOVE_TARGET_FOES_AND_ALLY` should be added.

File: test/battle/ability/parental_bond.c [hunk 3]
```diff
@@ -24,23 +24,166 @@ SINGLE_BATTLE_TEST("Parental Bond converts Tackle into a two-strike move")
...
+        EXPECT_EQ(player->species, SPECIES_KANGASKHAN_MEGA);
+    }
+}
+
+DOUBLE_BATTLE_TEST("Parental Bond does not convert multi-target moves into a two-strike move in Double Battles")
+{
+    u16 move;
+    PARAMETRIZE { move = MOVE_EARTHQUAKE; }
+    PARAMETRIZE { move = MOVE_ROCK_SLIDE; }
+    GIVEN {
```
Comment (AlexOn1ine) [inline test/battle/ability/parental_bond.c:79]: Assume for `MOVE_TARGET_FOES_AND_ALLY`

File: test/battle/ability/parental_bond.c [hunk 4]
```diff
@@ -24,23 +24,166 @@ SINGLE_BATTLE_TEST("Parental Bond converts Tackle into a two-strike move")
...
+        EXPECT_EQ(playerLeft->species, SPECIES_KANGASKHAN_MEGA);
+    }
+}
+
+DOUBLE_BATTLE_TEST("Parental Bond does not convert multi-target moves into a two-strike move in Double Battles, even if it only damages one")
+{
+    GIVEN {
+        ASSUME(gBattleMoves[MOVE_EARTHQUAKE].split != SPLIT_STATUS);
+        ASSUME(gBattleMoves[MOVE_EARTHQUAKE].strikeCount < 2);
+        ASSUME(gBattleMoves[MOVE_EARTHQUAKE].effect == EFFECT_EARTHQUAKE);
```
Comment (AlexOn1ine) [inline test/battle/ability/parental_bond.c:122]: Again maybe remove effect EQ and add `MOVE_TARGET_FOES_AND_ALLY`

**Review by AlexOn1ine — CHANGES_REQUESTED** [review]
Could you also check a normal double target move like Icy wind so both target conditions are tested?

**Comment by Bassoonian** [comment]
> Could you also check a normal double target move like Icy wind so both target conditions are tested?

That's why Rock Slide is there :)

**Comment by AlexOn1ine** [comment]
> > Could you also check a normal double target move like Icy wind so both target conditions are tested?
> 
> That's why Rock Slide is there :)

Oh sorry, mb. Though I would also remove the flinch effect assume.

**Comment by Bassoonian** [comment]
> > > Could you also check a normal double target move like Icy wind so both target conditions are tested?
> > 
> > 
> > That's why Rock Slide is there :)
> 
> Oh sorry, mb. Though I would also remove the flinch effect assume.

The flinch assume is necessary in this case, since it does in fact make the two opponents flinch (and the messages depend on that). I could alternatively move the flinch check there and make it check for regular Celebrate usage if the effect is not set to flinching, though

**Comment by AlexOn1ine** [comment]
> > > > Could you also check a normal double target move like Icy wind so both target conditions are tested?
> > > 
> > > 
> > > That's why Rock Slide is there :)
> > 
> > 
> > Oh sorry, mb. Though I would also remove the flinch effect assume.
> 
> The flinch assume is necessary in this case, since it does in fact make the two opponents flinch (and the messages depend on that). I could alternatively move the flinch check there and make it check for regular Celebrate usage if the effect is not set to flinching, though

In this case I would check the HP bar a second time but negated instead of the messages and celebrate.

**Comment by DizzyEggg** [comment]
What's the status of this? 👀

**Comment by AlexOn1ine** [comment]
superseded by #3973

**Pull request closed without merging**